### PR TITLE
[SPARK-33976][SQL][DOCS][FOLLOWUP] Fix syntax error in select doc page

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Add doc about `TRANSFORM` and related function.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Not need

### CODE REVIEW

File: docs/sql-ref-syntax-qry-select-transform.md [hunk 1]
```diff
@@ -51,11 +51,11 @@ row_format:
 
 * **row_format**    
 
-    Otherwise, uses the `DELIMITED` clause to specify the native SerDe and state the delimiter, escape character, null character and so on.
+    Uses `SERDE` to specify a custom SerDe to process inputs and outputs or uses the `DELIMITED` clause to specify the native SerDe and state the delimiter, escape character, null character and so on.
```
Comment (AngersZhuuuu) [inline docs/sql-ref-syntax-qry-select-transform.md:54]: Here update to contains two mode.

File: docs/sql-ref-syntax-qry-select-transform.md [hunk 2]
```diff
@@ -51,11 +51,11 @@ row_format:
...
 
 * **SERDE**
 
-    Specifies a custom SerDe for one table.
+    Specifies a custom SerDe to process inputs and outputs.
```
Comment (AngersZhuuuu) [inline docs/sql-ref-syntax-qry-select-transform.md:59]: Here should not be a table, we use `SerDe` to process data

**Review by AngersZhuuuu — COMMENTED** [review]
FYI @maropu

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan @maropu 
When I back port #31010  to branch 3.0/3.1 I noticed that here's syntax is wrong. and I have modify this in  https://github.com/apache/spark/pull/32375 and https://github.com/apache/spark/pull/32376

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Merged to master.

**Pull request closed without merging**